internal/console/console.go

```diff
@@ -0,0 +1,105 @@
+package console
+
+import (
+	"context"
+
+	api "github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/internal/store"
+	"github.com/flightctl/flightctl/internal/tasks"
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+)
+
+const ChannelSize = 2048
+
+type ConsoleSession struct {
+	UUID       string
+	OrgId      uuid.UUID
+	DeviceName string
+	SendCh     chan []byte
+	RecvCh     chan []byte
+}
+
+type InternalSessionRegistration interface {
+	// Register a session with a given UUID and channels
+	// Those channels will be used to read from and write to the session
+	// in a way that this interface down to the gRPC session is abstracted
+	StartSession(*ConsoleSession) error
+	CloseSession(*ConsoleSession) error
+}
+
+type ConsoleSessionManager struct {
+	store           store.Store
+	log             logrus.FieldLogger
+	callbackManager tasks.CallbackManager
+	configStorage   tasks.ConfigStorage
+	// This one is the gRPC Handler of the agent for now, in the next iteration
+	// this should be split so we funnel traffic through a queue in valkey
+	sessionRegistration InternalSessionRegistration
+}
+
+func NewConsoleSessionManager(store store.Store, callbackManager tasks.CallbackManager, configStorage tasks.ConfigStorage, log logrus.FieldLogger, sessionRegistration InternalSessionRegistration) *ConsoleSessionManager {
+	return &ConsoleSessionManager{
+		store:               store,
+		log:                 log,
+		callbackManager:     callbackManager,
+		configStorage:       configStorage,
+		sessionRegistration: sessionRegistration,
+	}
+}
+
+func (m *ConsoleSessionManager) StartSession(ctx context.Context, orgId uuid.UUID, deviceName string) (*ConsoleSession, error) {
+
+	session := &ConsoleSession{
+		OrgId:      orgId,
+		DeviceName: deviceName,
+		UUID:       uuid.New().String(),
+		SendCh:     make(chan []byte, ChannelSize),
+		RecvCh:     make(chan []byte, ChannelSize),
+	}
+	// TODO(majopela): This still signals console session creation through an annotation on the device
+	// we should move this to a separate table in the database
+	_, err := m.store.Device().Get(ctx, orgId, deviceName)
+	if err != nil {
+		return nil, err
+	}
+
+	annotations := map[string]string{api.DeviceAnnotationConsole: session.UUID}
+	if err := m.store.Device().UpdateAnnotations(ctx, orgId, deviceName, annotations, []string{}); err != nil {
+		return nil, err
+	}
+
+	// tell the gRPC service, or the message queue (in the future) that there is a session waiting, and provide
+	// the channels to read and write to the websocket session
+	if err := m.sessionRegistration.StartSession(session); err != nil {
+		// if we fail to register the session we should remove the annotation (best effort)
+		deleteAnnotations := []string{api.DeviceAnnotationConsole}
+		_ = m.store.Device().UpdateAnnotations(ctx, orgId, deviceName, map[string]string{}, deleteAnnotations)
```

Comment:
should we at least log this error?

Reply:
this one if happens and the one at the top of the if.